.changeset/rude-years-allow.md

```diff
@@ -1,5 +1,63 @@
 ---
-"@itwin/presentation-components": minor
+"@itwin/presentation-components": major
 ---
 
-Add interactive and more detailed informational messages in tree and instance filter components.
+Tree: Added interactive and more detailed informational messages in the tree and its hierarchy level filtering components:
+
+- When a hierarchy level size exceeds given limit, a message is displayed, suggesting the results should be filtered to reduce the result set.
+- The hierarchy level filtering dialog informs whether provided filters reduce the result set to a small enough size to be displayed in the tree.
+
+Includes 2 breaking `@beta` API changes:
+
+- `PresentationTreeNodeRenderer` now takes `onClearFilterClick` and `onFilterClick` callback props with node identifier argument rather than `PresentationTreeNodeItem`. This was a necessary change to allow opening filtering dialog for a parent node from its child node. To react to this breaking change:
```

Comment:
I think this breaking change should be placed after `useHierarchyLevelFiltering`. In that case it's easier to react to `PresentationTreeNodeRenderer` breaking changes. Or we could provide different examples that are not based on `useHierarchyLevelFiltering`:

Before
```tsx
const [clickedNode, setClickedNode] = useState<PresentationTreeNodeItem>();
return <PresentationTreeNodeRenderer
  {...nodeProps}
  onFilterClick={(node: PresentationTreeNodeItem) => {
    setClickedNode(node);
  }}
/>;
```

After
```tsx
const [clickedNode, setClickedNode] = useState<PresentationTreeNodeItem>();
return <PresentationTreeNodeRenderer
  {...nodeProps}
  onFilterClick={(nodeId: string) => {
    const node = modelSource.getModel().getNode(nodeId);
    if (isTreeModelNode(node) && isPresentationTreeNodeItem(node.item)) {
      setClickedNode(node.item);
    }
  }}
/>;
```

Reply:
applied the suggestion